### PR TITLE
feat: add capability for insert multiple rows in single query

### PR DESCRIPTION
**Pull request by samarpanB**
Signed-off-by: Samarpan Bhattacharya <this.is.samy@gmail.com>

Adds new createAll method to support insert multiple rows in single query in memory connector
Enables multiInsertSupport in memory connector
Adds capability to createAll in dao.js for models to do multiple insert in one query via the supported connectors

Fixes https://github.com/loopbackio/loopback-connector/issues/227
See also https://github.com/loopbackio/loopback-next/issues/3357

## Checklist

- [x] [Sign off your commits](https://loopback.io/doc/en/contrib/code-contrib.html) with DCO (Developer Certificate of Origin)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [x] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

### CODE REVIEW

**Comment by samarpanB** [comment]
Added persistence hooks test cases as well. While adding those, I found some issues, fixed them as well. Please review @dhmlau @achrinza

**Comment by samarpanB** [comment]
loopback-connector release is needed before it can be merged. However, this can be reviewed until then.

**Comment by samarpanB** [comment]
Updated connector version after rebase.

**Comment by samarpanB** [comment]
@raymondfeng @achrinza @dhmlau Please review.

**Review by dhmlau — APPROVED** [review]
@samarpanB, thanks for the PR. Your changes look reasonable to me. I have 2 minor comments, please review. 

Before merging this PR, I'd like to get @raymondfeng or one more maintainer to approve. Thanks.

**Review by achrinza — COMMENTED** [review]
Other than @dhmlau's remarks, everything else LGTM :+1:

File: test/persistence-hooks.suite.js
Comment (samarpanB) [reply test/persistence-hooks.suite.js:685]: Yes, thats how the hooks used to work previously as well, when inserting multiple records in create method. For each entry in the array of records to be inserted, `before save` and `after save` hooks independently. SO as many records, that many hooks triggered. I kept the behaviour same. There are some data validation logic which is useful if triggered for each entry separately.

File: test/persistence-hooks.suite.js
Comment (samarpanB) [reply test/persistence-hooks.suite.js:689]: Replied above

**Comment by samarpanB** [comment]
@dhmlau I replied to your comment. Can you please check if I need to change the behaviour ?

**Comment by dhmlau** [comment]
@samarpanB, thanks for your response. LGTM. Since I have the "go ahead" from @achrinza as well, I'll be merging this PR. Thanks!

**Pull request merged**
